### PR TITLE
Code refactoring done as per mentors request

### PR DESCRIPTION
**Pull request by nas-cam**
- global.spec.js file added ---->please check if is it in the right place----> (when tests are executed, it doesn't save any data in folder- I'll look into it more)
- config.js added with configuration for API
- console.error gives relevant information
-jasmine.json executes tests
- run-tests.js file added and now tests can be executed based on controller or all of them

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+export const config = {
+    user: {
+        getUser: {
+            name: 'getUser',
```
Comment (elvedinsuljic-abh) [inline config.js:4]: No need for this name: 'getUser' key-pair, we already know from paren object what is this related to.

Comment (elvedinsuljic-abh) [inline config.js]: No need for this name: 'getUser' key-pair, we already know from paren object what is this related to.

File: config.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            method: 'GET',
+            url: 'store/inventory'
+        },
+        deleteOrderById: {
+            name: 'deleteOrderById',
+            method: 'DELETE',
+            url: (orderId) => `store/order/${orderId}`
+        }
+    }
+}
```
Comment (elvedinsuljic-abh) [inline config.js]: New line should be inserted here, because of this "-" sign.

File: API-Automation/tasks/api-tasks.js [hunk 3]
```diff
@@ -1,173 +1,152 @@
...
-    return await axiosInstance.get(getUserPath.path);
+    return await axiosInstance.get(config.user.getUser.url(username));
   } catch (error) {
-    console.error(error);
+    console.error("Error message: ", error.response.data.message);
   }
 }
 
-export async function createUser(newUserData) {
+export async function createUser(data) {
```
Comment (elvedinsuljic-abh) [inline API-Automation/tasks/api-tasks.js]: Let's be precise here, what is argument "data"? Pls apply to all tasks.

File: API-Automation/tasks/api-tasks.js [hunk 4]
```diff
@@ -1,173 +1,152 @@
...
 
-export async function getUserByUsername() {
+
+export async function getUserByUsername(username) {
   try {
-    return await axiosInstance.get(getUserPath.path);
+    return await axiosInstance.get(config.user.getUser.url(username));
   } catch (error) {
-    console.error(error);
+    console.error("Error message: ", error.response.data.message);
```
Comment (elvedinsuljic-abh) [inline API-Automation/tasks/api-tasks.js]: Good job! :)

File: API-Automation/tasks/api-tasks.js [hunk 5]
```diff
@@ -1,173 +1,152 @@
...
-    return await axiosInstance.get(userLoginPath.path);
+    return await axiosInstance.get(config.user.userLogin.url(username, password));
   } catch (error) {
-    console.error(error);
+    console.error("Error message: ", error.response.data.message);
   }
 }
 
-export async function loginWithUsernameOnly() {
+export async function loginWithUsernameOnly(username) {
```
Comment (elvedinsuljic-abh) [inline API-Automation/tasks/api-tasks.js]: See, it is easier when we know what is the data, i.e. username

File: API-Automation/test-suites/global.spec.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+console.log("global.spec fajl is executed");
+
+if (typeof module !== 'undefined' && module.exports) {
+  // Kod se izvršava u Node.js okruženju
```
Comment (elvedinsuljic-abh) [inline API-Automation/test-suites/global.spec.js]: No need for comments like this, and down below.

File: API-Automation/test-suites/regression/pet/get_request.spec.js [hunk 7]
```diff
@@ -6,26 +6,26 @@ describe('Search order', () => {
...
     it('TC_10_2 User is able to find pet by Id', async () => {
-        const response = await findPetById();
+        const response = await findPetById(petId);
         expect(response.status).toBe(200);
         expect(_.isEqual(response.data, newPetData));
     });
-    
+
     afterEach(async () => {
         if (petId) {
```
Comment (elvedinsuljic-abh) [inline API-Automation/test-suites/regression/pet/get_request.spec.js]: In all afterEach() methods, simply call await deletePet, deleteUser, etc. No need for checking if petId exists, if it does not exist, test will fail in it() block also.

**Review by elvedinsuljic-abh — CHANGES_REQUESTED** [review]
Overall, good job with config files, it is looking much cleaner now.

**Review by elvedinsuljic-abh — CHANGES_REQUESTED** [review]
Overall, good job with config files, it is looking much cleaner now.

File: config.js (hunk 2 above)
Comment (nas-cam) [reply config.js]: New line inserted.

File: API-Automation/tasks/api-tasks.js (hunk 5 above)
Comment (nas-cam) [reply API-Automation/tasks/api-tasks.js]: Yes, I see. It is more intuitive when we know what the data is.

File: API-Automation/test-suites/regression/pet/get_request.spec.js (hunk 7 above)
Comment (nas-cam) [reply API-Automation/test-suites/regression/pet/get_request.spec.js]: Ok. I didn't think that way, unfortunately :(
I made the changes.

**Review by elvedinsuljic-abh — APPROVED** [review]
LGTM :)

**Pull request merged**